### PR TITLE
Isolating flags for cofix/fix reduction + adjusting names of reduction functions to what they do

### PR DESCRIPTION
**Pull request by herbelin**
This commit implements two things:
- first giving specific names to fix and cofix reductions as discussed already in May 2010; I used phi and psi, because I'm more used to see mu and nu as fixpoint/cofixpoint on types but this is a bit arbitrary
- second renaming the functions in the code so that they match what they do; in particular whd_betadeltaiota is renamed to whd_all, making explicit that it includes zeta on the contrary of whd_betadeltaiota

Of course, the drawback is that this would require changes in probably many third-party code.

So, to summarize, the question is whether this is perceived as a progress or a source of pain for third-party code updating. Maybe, compromises are possible.

Note that the commit is done so that no function keeps the same name with its semantics changing.

Nothing is done at the user-level, where we might need to keep iota to mean also reduction of fix/cofix for compatibiltiy purpose.

### CODE REVIEW

**Comment by herbelin** [comment]
I'm leaning to think that the second part of this commit is a progress in the sense of cleaning the API, giving names to reduction functions which correspond to what they do (I'm thinking in particular to whd_betadeltaiota which was doing zeta without telling it in its name), moreover without a function keeping the same name with a different behavior.

For the second part, which introduces reductions phi and psi, I tend to think that changing the semantics of iota (which would now not do phi and psi) is not good. So, maybe should we introduce, say iota0, or iotacore, or iotabase, such that iota becomes a short-hand for iota0 + phi + psi, with a strict compatibility.

**Comment by aspiwack** [comment]
Having iota standing for some iota0+phi+psi feels a bit hackish. But on the other hand, it will require less adaptation for users (plus, to be honest, this combination is often used together). That's fair enough.

Changing the name of functions in Ocaml is only a problem for the authors of plugins which are meant to span several Coq version (like @silene if I remember correctly). What do they think?

**Comment by mattam82** [comment]
I like the cleaner API even if it means adapting ML code and the renaming make sense to me. I also agree with the pragmatic view of giving a new name to iota without phi and psi at the user level.

**Comment by gares** [comment]
vconv.ml seem to contain a spurious merge conflict committed in.

**Comment by maximedenes** [comment]
Following discussion, I reimplemented it as #231

**Comment by maximedenes** [comment]
Merged as #231.
